lib/internal/Magento/Framework/App/FeedFactory.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Framework\App;
+
+use Magento\Framework\App\FeedFactory;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Feed factory
+ */
+class FeedFactory implements \Magento\Framework\App\FeedFactoryInterface
+{
+    /**
+     * @var \Zend_Feed
+     */
+    private $feedProcessor;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param \Zend_Feed $feedProcessor
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        \Zend_Feed $feedProcessor,
+        LoggerInterface $logger
+    ) {
+        $this->feedProcessor = $feedProcessor;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Get a new \Magento\Framework\App\Feed object from a custom array
+     *
+     * @throws \Magento\Framework\Exception\RuntimeException
```

Comment:
Sorry I have to disagree on this one as here we are catching all kinds of errors that can happen during the process, some of them are Zend_Feed exceptions, so it's not just input errors.